util/a2e/atoe_utils.c

```diff
@@ -545,22 +553,29 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					/* ibm@8665
 					 * Add floating point support for dbl2str & flt2str
 					 */
-					char *b;
-					int len;
+					char *tempPattern;
+					size_t freeCap;
+					double argument;
+					int requiredSpace;
 
-					b = a2e((char *)pattern, fmt - pattern);
+					tempPattern = a2e((char *)pattern, fmt - pattern);
```

Comment:
Reorder the declarations so each has an initializer:
```c
					char *tempPattern = a2e((char *)pattern, fmt - pattern);

					/* Extract a double from args, this works for both doubles
					 * and floats,
					 * NB if we use float for a single precision floating
					 * point number the result is wrong.
					 */
					size_t freeCap = (this.end > this.buffer) ? (this.end - this.buffer) : 0;
					double argument = va_arg(args, double);
					int requiredSpace = snprintf(this.buffer, freeCap, tempPattern, argument);
```
Note the added parentheses for `freeCap`.